File: 10.Math数学/leetcode_1401_checkOverlap.py
class Solution:
    def checkOverlap(self, radius: int, xCenter: int, yCenter: int, x1: int, y1: int, x2: int, y2: int) -> bool:
        # x∈[x1,x2], y∈[y1,y2], (x-xCenter)**2+(y-yCenter)**2最小
        x = x1 if xCenter<x1 else (x2 if x2<xCenter else xCenter)
        y = y1 if yCenter<y1 else (y2 if y2<yCenter else yCenter)
        return (xCenter-x)**2 + (yCenter-y)**2 <= radius*radius

if __name__ == "__main__":
    s = Solution()
    radius, xCenter, yCenter = 1, 0, 0
    x1, y1, x2, y2 = 1, -1, 3, 1
    print(s.checkOverlap(radius, xCenter, yCenter, x1, y1, x2, y2))

File: 10.Math数学/leetcode_1401_checkOverlap.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool checkOverlap(int radius, int xCenter, int yCenter, int x1, int y1, int x2, int y2) {
        int x = (xCenter<x1 ? x1 : (xCenter>x2 ? x2 : xCenter));
        int y = (yCenter<y1 ? y1 : (yCenter>y2 ? y2 : yCenter));
        return (x-xCenter)*(x-xCenter) + (y-yCenter)*(y-yCenter) <= radius*radius;
    }
};

int main(){
    Solution s;
    int radius = 1, xCenter = 0, yCenter = 0;
    int x1 = 1, y1 = -1, x2 = 3, y2 = 1;
    cout << s.checkOverlap(radius, xCenter, yCenter, x1, y1, x2, y2) << endl;
}
